client/src/modules/authorization/components/SignUpForm.jsx

```diff
@@ -0,0 +1,319 @@
+import React, { Fragment, PureComponent } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import _find from 'lodash/find';
+import validator from 'validator';
+
+import SignUpInput from './SignUpInput';
+import { signUp } from 'modules/authorization/model/actions';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import Preloader, {
+  PreloaderSize,
+  PreloaderTheme
+} from 'common/components/Preloader';
+
+class SignUpForm extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      confirmationSend: false,
+      email: '',
+      errors: { nameError: '', emailError: '', passwordError: '' },
+      isEmailValid: false,
+      isFormValid: false,
+      isNameValid: false,
+      isPasswordConfirmValid: false,
+      isPasswordValid: false,
+      name: '',
+      password: '',
+      passwordConfirm: '',
+      pending: false
+    };
+  }
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onNameChange = (name, isValid) => {
+    this.setState({ name, isNameValid: isValid });
+  };
+
+  onEmailChange = (email, isValid) => {
+    this.setState({ email, isEmailValid: isValid });
+  };
+
+  onPasswordChange = (password, isValid) => {
+    this.setState(
+      { password, isPasswordValid: isValid },
+      this.comparePasswords
+    );
+  };
+
+  onPasswordConfirmChange = (passwordConfirm, isValid) => {
+    this.setState(
+      { passwordConfirm, isPasswordConfirmValid: isValid },
+      this.comparePasswords
+    );
+  };
+
+  nameValidator = value => {
+    const emptyInfoId = 'authorization.input.username.empty';
+    const errorInfoId = 'authorization.input.username.invalid';
+    const { isEmpty, isLength } = validator;
+
+    if (isEmpty(value)) {
+      return emptyInfoId;
+    }
+
+    if (!isLength(value, { min: 1, max: 32 })) {
+      return errorInfoId;
+    }
+
+    return '';
+  };
+
+  emailValidator = value => {
+    const emptyInfoId = 'authorization.input.email.empty';
+    const errorInfoId = 'authorization.input.email.invalid';
+    const { isEmpty, isEmail } = validator;
+
+    if (isEmpty(value)) {
+      return emptyInfoId;
+    }
+
+    if (!isEmail(value)) {
+      return errorInfoId;
+    }
+
+    return '';
+  };
+
+  passwordValidator = value => {
+    const emptyInfoId = 'authorization.input.password.empty';
+    const errorInfoId = 'authorization.input.password.invalid';
+    const { isEmpty, isLength } = validator;
+
+    if (isEmpty(value)) {
```

Comment:
no trimming?

Reply:
This validator method is passed as a prop to SignUpInput and it's used there in the handleChange method. In this method, it receives and uses a trimmed value.